tests/tests_app/conftest.py

```diff
@@ -46,6 +47,13 @@ def pytest_sessionfinish(session, exitstatus):
             continue
         child.kill()
 
+    main_thread = threading.current_thread()
+    for t in threading.enumerate():
+        if t is not main_thread:
+            t.join(0)
+
+    Popen(f"sudo py-spy top --pid {os.getpid()}", shell=True)
```

Comment:
Could you please add a comment what this does? And what does this mean if I run it locally, it will ask for my password. And secondly, what if py-spy is not installed (I don't have it my my current lightning env). This looks like it will just silently fail, is this intentional?

Reply:
Removed.